### ISSUE TITLE
Missing `work done progress token` from client init

### ISSUE THREAD

**Issue by vilelamarcospaulo** [neovim/neovim#27938]
### Problem

Right now, `clojure-lsp` relies on the microsoft lsp specification that delegates the `Work Done progress token` creation to the sender. 

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initiatingWorkDoneProgress
> Work Done progress can be initiated in two different ways:
> 
> 1. by the sender of a request (mostly clients) using the predefined workDoneToken property in the requests parameter literal. The document will refer to this kind of progress as client initiated progress.

The problem is their implementation ignores the `$/progress` publication when this is not present, this way none `lsp-status` tool  works correctly to give to user feedback, what is bad since we don't really know when the processing is finished and can take a while. 
Others LSP (luals, golsp, etc...) just sets a random value `(string | int)` when it's not present and everything is working.

But in this case (maybe other LSP also relies in this definitions) it's just is ignored. 

To resolve this, in my local neovim build updated the `function Client:initialize()` implementation to also sent the property `workDoneToken = "foobar"` inside the `initialize_params`. Since this is the same approach used by `vscode` and `emacs`.
After this change the `$/progress` starts to receive every progress update, thus `lsp-status` and `fidget` was working just fine.

### Steps to reproduce using "nvim -u minimal_init.lua"

```
vim.lsp.set_log_level("debug")
vim.lsp.start_client({
  name = 'my-server-name',
  cmd = { 'clojure-lsp' },
  root_dir = vim.fn.getcwd()
})

```

### Expected behavior

should contains the `$/progress` in the lsp debug logs

```
-- [DEBUG][2024-03-19 18:26:22] .../vim/lsp/rpc.lua:403	"rpc.receive"	{  jsonrpc = "2.0",  method = "$/progress",  params  = {    token = "foobar",    value = {      kind = "begin",      percentage = 0,      title = "clojure-lsp"    }  }}
-- [DEBUG][2024-03-19 18:26:22] .../vim/lsp/rpc.lua:403	"rpc.receive"	{  jsonrpc = "2.0",  method = "$/progress",  params = {    token = "foobar",    value = {      kind = "report",      message = "Finding kondo config",      percentage = 5    }  }}
```

### Neovim version (nvim -v)

Nvim 0.9.5

### Language server name/version

clojure-lsp/2024.02.01-11.01.59

### Operating system/version

macos/14.2.1  (sonoma)

**Comment by rudiejd** [comment]
should this be a config parameter for each language server, or should Neovim always create the progress token itself? Are there downsides to creating / not creating the token as the client?

**Comment by vilelamarcospaulo** [comment]
Hello @rudiejd  
Looking at other implementations, there basic two scenarios for the `language servers`. Or they expect for the server to send this like the `clojure-lsp` case, or  they ignore the one sent and create a new every time like `goplp` and `lua_ls`.

From what i saw, when the server creates it by itself this creates no effect on the expected behavior. But for the ones where this is expected this change will enables to fully work.

Since this is expected based on the specification for LSP in general, i don't believe it should be for each language server, but  a implementation for the protocol itself. [VSCode](https://github.com/microsoft/vscode-languageserver-node/blob/e1345cca06095d61d3c5c0ff961472d55cbcf798/client/src/common/client.ts#L1226) for example always send a generated UUID in the init call. [Emacs](https://github.com/emacs-lsp/lsp-mode/blob/71a5eb3d4a1f30e5068cdf9aad7dcad543e79f4c/lsp-mode.el#L7739) in the other hand just send a fixed string "1".

Bottom line is, receive this parameter is already expected from the LSPs since it's part of the specification. Some servers will ignore it and others will use the received one. So theres no real downside in this.

**Comment by vilelamarcospaulo** [comment]
@rudiejd Thanks for doing this  implementation, after taking a fast look it looks pretty much ok to me. Right now i'm also working with a fixed string in my local build, and it's just fine, so i believe sending fixed "1" as emacs will be the best choice.
